stm32-modules/thermocycler-refresh/simulator/lid_heater_thread.cpp

```diff
@@ -12,10 +12,16 @@ using namespace lid_heater_thread;
 class SimLidHeaterPolicy {
   private:
     double _power = 0.0F;
+    periodic_data_thread::PeriodicDataThread *_periodic_data;
```

Comment:
Is there a way we can get away from raw pointers here? It doesn't really give us that much since these objects have equivalent lifetime but it might save us some tears in the future if we start segfaulting while shutting down or something. I guess they might have to be weak_ptrs since I think there's circular refs here though.

Reply:
Yep I can shift things around to get a `shared_ptr` in here